File: src/main/java/com/blink/backend/domain/model/appointment/Appointment.java
package com.blink.backend.domain.model.appointment;

import com.blink.backend.persistence.entity.auth.Users;
import com.blink.backend.persistence.entity.clinic.Clinic;

import java.time.LocalDateTime;

public class Appointment {

    private Long id;

    private Patient patient;

    private LocalDateTime scheduledTime;

    private Clinic clinic;

    private Integer duration;

    private AppointmentStatus appointmentStatus;

    private ServiceType serviceType;

    private String notes;

    private Users attendedByUser;

    private LocalDateTime attendedAt;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

}

File: src/main/java/com/blink/backend/domain/model/Appointment.kt
package com.blink.backend.domain.model

import com.blink.backend.persistence.entity.appointment.AppointmentStatus
import java.time.LocalDateTime

data class Appointment(
    var code: String? = null,
    val patient: Patient,
    val clinic: Clinic,
    val scheduledTime: LocalDateTime? = null,
    val notes: String? = null,
    val status: AppointmentStatus? = null,
    val duration: Int? = null,
    val scheduledTimeEnd: LocalDateTime? = null,
) {

    fun updateAppointmentDuration(duration: Int): Appointment {
        return this.copy(
            duration = duration,
            scheduledTimeEnd = scheduledTime?.plusMinutes(duration.toLong())
        )
    }
}
